### ISSUE TITLE
Coordinates of output incorrect.

### ISSUE THREAD

**Issue by dgriffiths3** [mapillary/OpenSfM#183]
Hi there,

I am having problems getting the outputted point cloud to be geo-referenced. From reading the docs I understand this is supposed to be a feature?

After running `opensfm_run_all` the `camera_models.json` file outputted is as follows:

```
{
    "v2 dji fc6310 4864 3648 perspective 0.6666": {
        "focal_prior": 0.6666666666666666, 
        "width": 4864, 
        "k1": 0.0, 
        "k2": 0.0, 
        "k1_prior": 0.0, 
        "k2_prior": 0.0, 
        "projection_type": "perspective", 
        "focal": 0.6666666666666666, 
        "height": 3648
    }
}
```

The `reference_lla.json` file is as follows:

`{"latitude": 51.67268379738563, "altitude": 0, "longitude": -0.13180468137254903}`

which is the correct latitude and longitude. The altitude would not be 0, but Im assuming there is more sense to that.

When looking into the `reconstruction.json` file, I do not recognise the GPS data. 

```
"shots": {
            "DJI_0019.JPG": {
                "orientation": 1, 
                "camera": "v2 dji fc6310 4864 3648 perspective 0.6666", 
                "gps_position": [
                    -61.359440189069325, 
                    20.383708417208254, 
                    1.9996728720143437
                ], 
                "gps_dop": 15.0, 
                "rotation": [
                    -2.654405958100366, 
                    -1.658051680557088, 
                    0.03950194648563577
                ], 
                "translation": [
                    8.501926443319572, 
                    63.969734316737046, 
                    1.3988852470038537
                ], 
                "capture_time": 1496588849.0
            }, 
```

When I do import the point cloud into a GIS using WGS84, the centre of the point cloud always has the coordinates of `0, 0`, and all of the height data are minus values.

As a test I have tried running the same images through Agisoft Photoscan to see if maybe it was an issue with the camera etc.. but it outputs a correctly geo-referenced point cloud.

Is there something I am missing here?

Many Thanks,

Dave

**Comment by paulinus** [comment]
Hi @dgriffiths3,

The coordinates for points, cameras and `gps_position` are stored in what we call [*world coordinates*](http://opensfm.readthedocs.io/en/latest/geometry.html#world-coordinates)

This is a topocentric reference frame local to this reconstruction. The [geo.py](https://github.com/mapillary/OpenSfM/blob/master/opensfm/geo.py) module offers functions to transform to ECEF coordinates or WGS84 lat, lons.

What coordinates would you like to have the point cloud written in?

It might make sense to add a command to export to other coordinate systems, or add coordinate system parameter to the `export_ply` command.

**Comment by dgriffiths3** [comment]
Hi @paulinus thanks for the reply.

I have had a look into the geo.py, but am struggling with a few steps. Firstly I was hoping to output my point cloud in `WGS 84 UTM31N`, although I would have no issue re-projecting so long as I know the output coordinate system.

I cannot understand where I would input the parameters for a specific ECEF reference system (i.e. `WGS 84 UTM31N`). I feel as if I am maybe missing something here? 

What I really cannot understand though is why every altitude has a minus value, all around -70 to the expected. I tried running the images through `OpenDroneMap` and this produced the same result, although eastings and northings were correct. Is this related to the fact that my `reference_lla.json` altitude is `0` ? All imagery was taken from UAV at approximately 120m, or is the 0 correct?

I have also tried just translating the altitude by a constant determined from a measured gcp, however this gave horrible results.

**Comment by dgriffiths3** [comment]
Hi @paulinus ,

I have discovered the reason for the incorrect altitude was due to the altitude recorded by the built in GPS being horribly underestimated. As the data was captured with a DJI drone the exif file also has a relative height tag (from take-off), therefore my solution is to update the altitude to `relative height - takeoff height`, which luckily I have a dGPS reading for.

Thanks for advise with the `geo.py` module.

Dave
